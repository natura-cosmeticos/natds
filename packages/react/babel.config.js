const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
