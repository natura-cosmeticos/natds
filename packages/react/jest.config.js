const base = require('../../jest.config.base.js')
const { displayName, name } = require('./package.json')

module.exports = {
  ...base,
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx}',
    '!src/**/*.stories.tsx',
    '!src/helpers/**'
  ],
  errorOnDeprecated: true,
  displayName,
  name,
  roots: ['<rootDir>/src'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx'
  }
}
