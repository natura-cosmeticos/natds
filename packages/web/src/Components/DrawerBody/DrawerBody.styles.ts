import createStyles from '../../styles/createStyles'
import { makeStyles } from '../../styles/makeStyles'
import { IThemeWeb } from '../../Themes'

type UseStylesArgs = {
  scrolled: boolean
}

export const boxShadow = ({ scrolled }: UseStylesArgs) => {
  if (scrolled) {
    return 'rgba(0, 0, 0, .14) inset 0 9px 5px -5px'
  }

  return 'none'
}

const style = ({ sizes }: IThemeWeb) => createStyles({
  root: {
    display: 'flex',
    flex: '1 1 auto',
    overflow: 'hidden',
    position: 'relative',
    '&::before': {
      boxShadow,
      content: '""',
      height: sizes?.small,
      left: 0,
      pointerEvents: 'none',
      position: 'absolute',
      right: 0,
      top: 0,

      /**
       * @todo replace hardcoded duration by time token
       */
      transition: 'box-shadow ease-in-out .1s',
      zIndex: 2
    }
  },
  scroll: {
    flex: '1 1 auto',
    overflowY: 'auto',
    padding: `${sizes?.none} ${sizes?.tiny}px`,
    position: 'relative'
  }
})

export const useStyles = makeStyles(style, { name: 'NatDSDrawerBody' })

export default useStyles
