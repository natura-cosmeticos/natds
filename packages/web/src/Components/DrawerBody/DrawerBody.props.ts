import * as React from 'react'

export interface IDrawerBodyProps {

  /**
   * The content of the component.
   *
   * @optional
   * @todo make property children required
   * @type node
   */
  children?: React.ReactNode;

  /**
   * The component used for the root node.
   *
   * Either a `string` to use or a HTML element or a component.
   *
   * @default "div"
   * @optional
   * @type element
   */
  component?: React.ElementType;

  /**
   * The component used for the scroll node.
   *
   * Either a `string` to use or a HTML element or a component.
   *
   * @default "div"
   * @since 0.24
   * @type element
   */
  scrollComponent?: React.ElementType
}
