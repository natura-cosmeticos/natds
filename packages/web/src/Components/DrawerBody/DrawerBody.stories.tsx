import { Meta, Story } from '@storybook/react'
// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { DrawerBody, IDrawerBodyProps } from './DrawerBody'
import { Navigation } from '../DrawerMenu/DrawerMenu.stories'
import { DrawerMenu } from '../DrawerMenu'

export default {
  component: DrawerBody,
  title: 'Components/Drawer/Drawer Body'
} as Meta

const Template : Story<IDrawerBodyProps> = (args: IDrawerBodyProps) => <DrawerBody {...args} />

export const Playground = Template.bind({})
Playground.args = {
  children: <DrawerMenu {...Navigation.args} />
}
