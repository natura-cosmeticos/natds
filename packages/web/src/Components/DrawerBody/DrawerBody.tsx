// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { IDrawerBodyProps } from './DrawerBody.props'
import { useStyles } from './DrawerBody.styles'

export { IDrawerBodyProps } from './DrawerBody.props'

/**
 * ## Importing
 *
 * ```
 * import { DrawerBody } from '@naturacosmeticos/natds-web';
 * ```
 */
export const DrawerBody = React.forwardRef<HTMLElement, IDrawerBodyProps>((
  props: IDrawerBodyProps, ref
) => {
  const {
    children,
    component: Component = 'div',
    scrollComponent: ScrollComponent = 'div',
    ...otherProps
  } = props
  const [
    scrolled, setScrolled
  ] = React.useState(false)

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const handleScrolled = (event: any) => {
    if (typeof event === 'undefined') {
      return
    }
    const { target } = event

    if (typeof target === 'undefined') {
      return
    }
    const { scrollTop } = target

    if (typeof scrollTop !== 'undefined') {
      const SCROLL_POSITION_ZERO = 0

      setScrolled(scrollTop > SCROLL_POSITION_ZERO)
    }
  }

  const { root, scroll } = useStyles({ scrolled })

  return (
    <Component
      className={root}
      ref={ref}
      {...otherProps}
    >
      <ScrollComponent
        className={scroll}
        onScroll={handleScrolled}
      >
        {children}
      </ScrollComponent>
    </Component>
  )
})

DrawerBody.displayName = 'DrawerBody'

export default DrawerBody
