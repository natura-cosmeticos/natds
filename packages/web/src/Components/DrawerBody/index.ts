import DrawerBody from './DrawerBody'

export { IDrawerBodyProps } from './DrawerBody.props'
export { DrawerBody }
export default DrawerBody
