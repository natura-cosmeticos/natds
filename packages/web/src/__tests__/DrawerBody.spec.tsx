import * as React from "react";
import * as renderer from "react-test-renderer";
import { expect } from "chai";
import { mount } from "enzyme";

import DrawerBody, { DrawerBodyComponent, DrawerBodyScrollComponent } from "../Components/DrawerBody";

describe("DrawerBody component", () => {
  test("should match snapshot - DrawerBody children", () => {
    const children = <div>some custom children</div>;
    const component = renderer.create(<DrawerBody>{children}</DrawerBody>);

    expect(component).matchSnapshot("DrawerBody with children snapshot");
  });

  test("should match snapshot - DrawerBody custom component", () => {
    const children = <div>some custom children</div>;
    const component = renderer.create(<DrawerBody component="section">{children}</DrawerBody>);

    expect(component).matchSnapshot("DrawerBody with custom component snapshot");
  });

  test("should custom children prevail over the properties", () => {
    const children = <div className="custom">some custom children</div>;
    const component = mount(<DrawerBody>{children}</DrawerBody>);
    const event: any = { target: { scrollTop: 10 } };

    component.find(DrawerBodyScrollComponent).simulate("scroll", event);

    expect(component.find(DrawerBodyComponent).prop("scrolled")).to.equal(true);
  });
});
