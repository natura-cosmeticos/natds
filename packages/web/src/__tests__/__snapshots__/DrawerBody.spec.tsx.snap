// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerBody component should match snapshot - DrawerBody children: DrawerBody with children snapshot 1`] = `
<div
  className="sc-AxjAm eaRsRM"
>
  <div
    className="sc-AxirZ ipcfTM"
    onScroll={[Function]}
  >
    <div>
      some custom children
    </div>
  </div>
</div>
`;

exports[`DrawerBody component should match snapshot - DrawerBody custom component: DrawerBody with custom component snapshot 1`] = `
<section
  className="sc-AxjAm eaRsRM"
>
  <div
    className="sc-AxirZ ipcfTM"
    onScroll={[Function]}
  >
    <div>
      some custom children
    </div>
  </div>
</section>
`;
